### ISSUE TITLE
High-throughput simulation for computational cell biology with Cloud-COPASI

### ISSUE THREAD

**Issue by hasanbaig** [nrnb/GoogleSummerOfCode#211]
### Background
[Cloud-COPASI](http://cloud-copasi-new.cam.uchc.edu) is a web-based tool for running computationally intensive simulation and analysis tasks with the COPASI simulator in parallel on high-performance computing pools. Cloud-COPASI can connect to existing computing pools, or alternatively launch a computing pool using Amazon’s Elastic Computing Cloud (EC2).

Cloud-COPASI can run a number of simulation and analysis tasks, including global sensitivity analyses, stochastic simulations, parameter scans, optimization, and parameter fitting. Each task is automatically split into a number of smaller jobs, which are executed in parallel, allowing for significant speed-ups in running time.
 

Models are prepared using the standalone COPASI application, a widely used free, open-source, cross-platform biochemical simulation tool.

Cloud-COPASI can connect to existing Condor, PBS, LSF, Sun/Oracle Grid Engine, and Slurm compute pools by using the [HTCondor](https://htcondor.org/) software package. In order to connect, Cloud-COPASI requires SSH access to a submit node running Linux on the compute pool.

### Goal
Current implementation of Cloud-COPASI requires a user to first configure SSH access to the submit node using a public/private keypair. This assumes that for a user to be able to use Cloud-COPASI they must have login access to such a pool. Alternatively, we would also like to provide simulation services on our own HPC center to users that may otherwise not have access to HPC facilities.

The goal of this project is therefore to create a customized version of Cloud-COPASI that already has a predefined HPC pool connection which will be used by all registered users of this version of Cloud-COPASI. A quota system can also be implemented to prevent abuse of this resource. 

The result will be a server that allows users to run computing intensive COPASI simulations in UConn (University of Connecticut) Health’s HPC facility without a need to create login accounts and cryptographic keys.

### Size and Length of Project
- **medium: 175 hours**
- **12 weeks**

### Skills
List skills/technologies that the student should be familiar with. Also tag the issue with these.

Essential skills: Python, Django, HTML, front-end/back-end development, SSH. 
Nice to have skills: CSS, Javascript, Experience with COPASI. 

### Public Repository
[Cloud-COPASI](https://github.com/copasi/cloud-copasi )

### Potential Mentors
Hasan Baig ([hbaig@uchc.edu](mailto:hbaig@uchc.edu)) 
Pedro Mendes ([pmendes@uchc.edu](mailto:pmendes@uchc.edu))

**Comment by osamaahmed17** [comment]
This project seems to be interesting for GSOC 2022 and it could bring my experience of **Cloud Computing** and **Software Development** into action. Looking forward to contributing to it.

**Comment by hasanbaig** [comment]
> This project seems to be interesting for GSOC 2022 and it could bring my experience of **Cloud Computing** and **Software Development** into action. Looking forward to contributing to it.

Thanks for showing your interest @osamaahmed17. Please have a look at the [Cloud-COPASI](http://cloud-copasi-new.cam.uchc.edu/) website and let us know if you have any queries regarding this. We have received your CV and will get back to you soon.

**Comment by khanhcodes** [comment]
My name is Kaitlyn and I am a first-year undergraduate student majoring in Computer Science and Mathematics at the University of Georgia. I am very interested in joining this project for the summer and making contributions to the code. In terms of website development, I am familiar with JavaScript, React.js, Node,js, HTML, CSS, and BootStrap. I am not that exported on Django, but I am willing to learn beforehand. For bioinformatics data analysis and backend, my skills/techniques include Python, Java, and some data analysis. Since I have experience both in front-end development and bioinformatics, I think my skill set would fit this project perfectly. I am interested in create a customized version of Cloud-COPASI in this project, as I think my skill set would fit this perfectly. I am very excited about the applications of computer science in biology, and I am looking forward to being a part of your team for the summer. Please let me know how I can get started!

**Comment by hasanbaig** [comment]
Hi Kaitlyn,
Thanks for your interest in this project. Please share your email address. I will send you the reading material which will help you understand the project better. 
Thanks.

**Comment by khanhcodes** [comment]
My email is knn53018@uga.edu. Please send me the material as I would love to get started working on this. Also, how many contributors do you generally accept for this project?

**Comment by khanspers** [comment]
Hi @khanhcodes: It's one contributor per project, and we won't know the total number of projects slots we are granted until after proposals are reviewed.

**Comment by khanspers** [comment]
A reminder that the application period opens on Monday April 4. Proposals to NRNB must be submitted on the official GSoC Site (https://summerofcode.withgoogle.com/) before April 19, 18:00 UTC to be considered, and contributors are encouraged to submit proposals in draft format early, so that mentors can give feedback directly at the GSoC site.

**Comment by AlexanderPico** [comment]
**IMPORTANT REMINDER: GSoC 2022 is for new “beginners” to open source.** 

Applicants are expected to review eligibility requirements prior to applying.  We can not accept applications from contributors with prior open source development experience. From the GSoC FAQ https://developers.google.com/open-source/gsoc/faq:

> Can someone already participating in open source be a GSoC Contributor?
>
> The goal of GSoC is to bring new contributors into open source organizations. GSoC can also help beginner contributors learn the ins and outs of open source while being mentored by experienced community members.
> GSoC is for new and beginner contributors to open source, it is not for experienced contributors to open source.

**Comment by khanspers** [comment]
Closing because this is an active project for GSoC 2022.
